adapters/rest-vertx/src/main/java/org/eclipse/hono/adapter/rest/VertxBasedRestProtocolAdapter.java

```diff
@@ -12,54 +12,94 @@
 
 package org.eclipse.hono.adapter.rest;
 
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.AuthProvider;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BasicAuthHandler;
 import org.eclipse.hono.adapter.http.AbstractVertxBasedHttpProtocolAdapter;
-import org.eclipse.hono.config.ServiceConfigProperties;
 
 import io.vertx.core.http.HttpMethod;
 import io.vertx.ext.web.Router;
-import io.vertx.ext.web.RoutingContext;
+import org.eclipse.hono.auth.UsernamePasswordCredentials;
+import org.eclipse.hono.config.ProtocolAdapterProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * A Vert.x based Hono protocol adapter for accessing Hono's Telemetry &amp; Event API using REST.
  */
-public final class VertxBasedRestProtocolAdapter extends AbstractVertxBasedHttpProtocolAdapter<ServiceConfigProperties> {
+public class VertxBasedRestProtocolAdapter
+        extends AbstractVertxBasedHttpProtocolAdapter<ProtocolAdapterProperties> implements AuthProvider {
 
-    private static final String PARAM_TENANT = "tenant";
+    private static final Logger LOG             = LoggerFactory.getLogger(VertxBasedRestProtocolAdapter.class);
+    private static final String PARAM_TENANT    = "tenant";
```

Comment:
Maybe we can avoid to have such right alignment for values ?

Reply:
done